includes/Widgets/Stories.php

```diff
@@ -0,0 +1,393 @@
+<?php
+/**
+ * Stories Widgets.
+ *
+ * @package Google\Web_Stories
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Widgets;
+
+use Google\Web_Stories\Story_Query;
+use WP_Widget;
+use function Google\Web_Stories\get_layouts;
+
+/**
+ * Class Stories
+ *
+ * @package Google\Web_Stories\Widgets
+ */
+class Stories extends WP_Widget {
```

Comment:
Crazy idea: instead of assembling the whole widget form in PHP and sprinkling some JS on top of it, would it make sense to use React as well? Probably not, but just want to get this thought out there :)

Reply:
I think we need to think about widget looks like when the block based widget areas go into core. Adding react here, might mean breakages. I also don't want our widget so different from core blocks.